test/tests/functional/pbs_stf.py

```diff
@@ -0,0 +1,761 @@
+        jid = self.server.submit(j)
+        self.server.expect(JOB, {'job_state': 'Q'}, id=jid)
+
+        a = {'queue': 'ded_time',
+              'Resource_List.max_walltime': '20:00:00',
+              'Resource_List.min_walltime': '00:01:00'}
+        j = Job(TEST_USER, attrs=a)
+        j2id = self.server.submit(j)
+        self.server.expect(JOB, {'job_state': 'R'}, id=j2id)
+        self.server.expect(JOB, {'job_state': 'Q'}, id=jid)
+
+        attr = {'Resource_List.walltime': (LE, '01:40:00')}
+        self.server.expect(JOB, attr, id=j2id)
+
+        attr = {'Resource_List.walltime': (GE, '00:01:00')}
+        self.server.expect(JOB, attr, id=j2id)
+
+        sw = self.server.status(JOB, 'Resource_List.walltime', id=j2id)
+        wt = sw[0]['Resource_List.walltime']
+        wt2 = wt
+
+        # Make walltime given by qstat to agree with format in sched_logs.
+        # A leading '0' is removed in the hour string.
+        hh = wt.split(':')[0]
+        if len(hh) == 2 and hh[0] == '0':
+            wt2 = wt[1:]
+
+        msg = "Job;%s;Job will run for duration=[%s|%s]" % (j2id, wt, wt2)
+        m = self.scheduler.log_match(
+            msg, regexp=True, starttime=(now - 1), max_attempts=5, interval=2)
+        self.assertTrue(len(m) > 0)
+
+    def test_t_4_1_1(self):
+        """
+        Test shrink to fit by setting a dedicated time that starts 1 hour
+        from now for 1 hour and strict_ordering is set to true. Submit a
+        regular job requesting 1 ncpu which goes into Q state.  The submit
+        a job that can run for as low as 10 minutes and as long as 10 hours.
+        Expect the first job to be in Q state and the second job
+        in R state with a walltime that is less than or equal to 1 hour
+        and greater than or equal to 10 minutes.
+        """
+        a = {'resources_available.ncpus': 1}
+        self.server.manager(MGR_CMD_SET, NODE, a, id=self.mom.shortname)
+
+        now = int(time.time())
+        self.scheduler.add_dedicated_time(start=now + 3600, end=now + 7200)
+        self.scheduler.set_sched_config({'strict_ordering': 'True'})
+
+        a = {'Resource_List.ncpus': '1'}
+        j = Job(TEST_USER, attrs=a)
+        jid = self.server.submit(j)
+        self.server.expect(JOB, {'job_state': 'Q'}, id=jid)
+
+        a = {'Resource_List.max_walltime': '10:00:00',
+              'Resource_List.min_walltime': '00:10:00'}
+        j = Job(TEST_USER, attrs=a)
+        j2id = self.server.submit(j)
+        self.server.expect(JOB, {'job_state': 'R'}, id=j2id)
+        self.server.expect(JOB, {'job_state': 'Q'}, id=jid)
+
+        attr = {'Resource_List.walltime': (LE, '01:00:00')}
+        self.server.expect(JOB, attr, id=j2id)
+
+        attr = {'Resource_List.walltime': (GE, '00:10:00')}
+        self.server.expect(JOB, attr, id=j2id)
+
+    def set_primetime(self, ptime_start, ptime_end):
+        # Delete all entries in the holidays file
+        self.scheduler.holidays_delete_entry('a')
```

Comment:
This method is defined in between which looks awkward. can you move at the beginning of the class.